datajunction-server/datajunction_server/database/user.py

```diff
@@ -48,6 +74,23 @@ class User(Base):
         Enum(OAuthProvider),
     )
     is_admin: Mapped[bool] = mapped_column(default=False)
+    kind: Mapped[PrincipalKind] = mapped_column(
+        Enum(PrincipalKind),
+        default=PrincipalKind.USER,
+    )
+
+    created_by_id: Mapped[int | None] = mapped_column(
+        ForeignKey("users.id"),
+        nullable=True,
+    )
+
+    created_at: Mapped[UTCDatetime | None] = mapped_column(
+        DateTime(timezone=True),
+        insert_default=partial(datetime.now, timezone.utc),
+        nullable=True,
+    )
+
+    created_by: Mapped["User"] = relationship("User")
```

Comment:
How is this different from `created_by_id` ?

Reply:
This is the ORM reference to the `User` object, similar to how `created_nodes` works on the user class as well.